geps/gep-3139/index.md

```diff
@@ -0,0 +1,208 @@
+
+`timeout.maxStreamDuration`
+
+- The timeout value to allow users to configure streaming duration. If a GatewayAPI implementation and a [gRPC header timeout](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#requests) are both present, the "stricter" of the values will take precedence . A value of `0s` will disable a timeout - meaning a client's request will not timeout. This field is optional. 
+
+`timeout.strictEnforcement`
+
+- An enum value for specifying between GRPCRoute timeouts and grpc-timeout metadata timeoutsm. If set to `allow`, the timeout value that is stricter of two. If set to `deny`, it is assumed the implementation does not allow for this features. 
+
+If no timeout values are specified in the GRPCRoute, then the `grpc-timeout` header (from the client) will determine a request's timeout.
+
+
+GO
+```
+type GRPCRouteRule struct {
+    // Timeouts defines the timeouts that can be configured for an gRPC request.
+    //
+    // Support: Extended
+    //
+    // +optional
+    // <gateway:experimental>
+    Timeouts *GRPCRouteTimeouts `json:"timeouts,omitempty"`
+
+    // ...
+}
+
+// GRPCRouteTimeouts defines timeouts that can be configured for an GRPCRoute.
+// Timeout values are represented with Gateway API Duration formatting.
+// Specifying a zero value such as "0s" is interpreted as no timeout.
+//
+// +kubebuilder:validation:XValidation:message="backendRequest timeout cannot be longer than request timeout",rule="!(has(self.request) && has(self.backendRequest) && duration(self.request) != duration('0s') && duration(self.backendRequest) > duration(self.request))"
+type GRPCRouteTimeouts struct {
+    // MaxStreamDuration specifies the maximum duration for upstream to respond to an gRPC request from a client.
+    // If upstream has not been able to respond before this deadline is met, upstream
+    // MUST return a timeout error.
+    //
+    // For example, setting the `rules.timeouts.MaxStreamRequest` field to the value `10s` in an
+    // `GRPCRoute` will cause a timeout if a client request is taking longer than 10 seconds
+    // to complete.
+    //
+    // This timeout is intended to cover as close to the whole request-response transaction
+    // as possible although an implementation MAY choose to start the timeout after the entire
+    // request stream has been received instead of immediately after the transaction is
+    // initiated by the client.
+    //
+    // When omitted, the timeout value will default to the `grpc-timeout` value in the request. 
+    //
+    // Support: Extended
+    //
+    // +optional
+    MaxStreamDuration *Duration `json:"maxStreamDuration,omitempty"`
+
+    // StrictEnforcement enum specifies which timeout to use request from a client 
+    // to upstream if 2 are specified. This occurs when a gRPC timeout is specified in the GRPCRoute  
```

Comment:
which 2?

Edit: I see the explanation after it, I think it makes sense just to say something like 
" StrictEnforcement enum specifies the behavior when a grpc-timeout header is provided, and the GRPCRoute configures a timeout...."

Reply:
Also, are Allow/Deny the right enums here? I am not sure if this should be something like "use from the metadata" vs "use from the configuration" vs "use the lowest" as part of configurable options.